api/v1/controllers/samples.js

```diff
@@ -196,25 +217,43 @@ module.exports = {
     const resultObj = { reqStartTime: new Date() };
     const reqStartTime = Date.now();
     const value = req.swagger.params.queryBody.value;
-    u.getUserNameFromToken(req,
-      featureToggles.isFeatureEnabled('enforceWritePermission'))
-    .then((userName) => {
+
+    if (featureToggles.isFeatureEnabled(constants.featureName)) {
       if (featureToggles.isFeatureEnabled('useWorkerProcess')) {
         const jobType = require('../../../jobQueue/setup').jobType;
         const jobWrapper = require('../../../jobQueue/jobWrapper');
 
         const wrappedBulkUpsertData = {};
         wrappedBulkUpsertData.upsertData = value;
-        wrappedBulkUpsertData.userName = userName;
         wrappedBulkUpsertData.reqStartTime = reqStartTime;
+        wrappedBulkUpsertData.isRedisEnabled = true;
 
         const j = jobWrapper.createJob(jobType.BULKUPSERTSAMPLES,
           wrappedBulkUpsertData, req);
       } else {
-        helper.model.bulkUpsertByName(value,
-          userName);
+        redisModelSample.bulkUpsertSampleInRedis(value);
```

Comment:
Can we change this to return the list of samples, to be used by the real-time publisher?